WordPress/Classes/Services/ReaderPostService.m

```diff
@@ -643,16 +644,13 @@ - (void)mergePosts:(NSArray *)posts
         [self deletePostsFromBlockedSites];
         readerTopic.lastSynced = [NSDate date];
 
-        if (!skippingSave) {
-            // performBlockAndWait here so we know our objects are saved before we call success.
-            [self.managedObjectContext performBlockAndWait:^{
-                [[ContextManager sharedInstance] saveContext:self.managedObjectContext];
-            }];
-        }
-        if (success) {
-            BOOL hasMore = ((postsCount > 0 ) && ([self numberOfPostsForTopic:readerTopic] < ReaderPostServiceMaxPosts));
-            success(postsCount, hasMore);
-        }
+        [[ContextManager sharedInstance] saveContext:self.managedObjectContext withCompletionBlock:^{
+            // Is called on main queue
+            if (success) {
+                BOOL hasMore = ((postsCount > 0 ) && ([self numberOfPostsForTopic:readerTopic] < ReaderPostServiceMaxPosts));
```

Comment:
Note here: `numberOfPostsForTopic` also requires to be executed within the MOC's gcd queue. (While the `completionBlock` is actually executed within the main queue)